x-pack/test/functional_execution_context/tests/server.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../ftr_provider_context';
+import { assertLogContains, isExecutionContextLog, ANY } from '../test_utils';
+
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export default function ({ getService }: FtrProviderContext) {
+  const retry = getService('retry');
+  const supertest = getService('supertest');
+
+  async function waitForStatus(
+    id: string,
+    statuses: Set<string>,
+    waitMillis: number = 10000
+  ): Promise<Record<string, any>> {
+    if (waitMillis < 0) {
+      expect().fail(`waiting for alert ${id} statuses ${Array.from(statuses)} timed out`);
+    }
+
+    const response = await supertest.get(`/api/alerting/rule/${id}`);
+    expect(response.status).to.eql(200);
+    const { status } = response.body.execution_status;
+    if (statuses.has(status)) return response.body.execution_status;
+
+    // eslint-disable-next-line no-console
+    console.log(
+      `waitForStatus(${Array.from(statuses)} for id:${id}): got ${JSON.stringify(
+        response.body.execution_status
+      )}, retrying`
+    );
```

Comment:
I think we should use the `log` service instead of `console.log`? `getService('log')`